app/services/JourneyRepository.scala

```diff
@@ -50,26 +50,37 @@ class KeystoreJourneyRepository extends JourneyRepository with FrontendServicesC
   }
 
   override def get(id: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[JourneyData]] = {
-      cache.fetchAndGetEntry[JourneyData](cache.defaultSource, cacheId, id)
+    fetchCache[JourneyData](id)
   }
 
   override def getV2(id: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[JourneyDataV2]] = {
-    cache.fetchAndGetEntry[JsValue](cache.defaultSource, cacheId, id).map(_.map(json =>
+    fetchCache[JsValue](id).map(_.map(json =>
       (json \ "config" \ "version").asOpt[Int] match {
         case Some(_) => json.as[JourneyDataV2]
         case None => convertToV2Model(json.as[JourneyData])
       }
     ))
   }
 
+  private def fetchCache[A](id: String)(implicit reads: Reads[A], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[A]] = {
+      for {
+        newCachedDoc <- cache.fetchAndGetEntry[A](cache.defaultSource, id, cacheId)
```

Comment:
Wonder if we should clarify the parameter names. If cacheId is only used in this location maybe it can be a local val called something more appropriate?

Reply:
Refactored to change the parameters names as cacheId and keyId, same names used by the library.  Didn't change the keyId to local variable as it is used in 2 methods and didn't want to duplicate it.